components/ruby/ruby-32/Makefile

```diff
@@ -56,11 +57,10 @@ RUBY_VER= $(word 1,$(RUBY_VER_WORDS)).$(word 2,$(RUBY_VER_WORDS))
 RUBY_LIB_VER= $(RUBY_VER)
 
 PROTORUBYDIR= $(PROTO_DIR)/usr/ruby/$(RUBY_VER)
-ifeq ($(MACH), i386)
-GNU_CPU=x86_64
-else
+ifeq ($(MACH), sparc)
 GNU_CPU=sparc
```

Comment:
What you did here is not aligned with #19992 you are referring to.

Reply:
Then please use some other variable and do not abuse `GNU_CPU`.

Reply:
Why, should I not use it? - I tought shared-macros is there for this purpose - see the manifest.

Reply:
Should I _really_ repeat what I wrote [here](https://github.com/OpenIndiana/oi-userland/pull/19992#issuecomment-2541207192)?

Reply:
Okay, I've now changed Makefile and manifest to use RUBY_CPU instead of GNU_CPU - okay now?

Reply:
Far better.  If you would want to make it even better you can try to explore how to make the sample-manifest to contain `RUBY_CPU` too (to make it as close as possible to the production manifest).

Please make sure the publish works okay, because I think it does not.

Reply:
I made sure that it pubslishes correctly now, but I don't have a clue how to manipulate the Makefile in order to change the sample-manifest.to use RUBY_CPU. Is this now ready to go?

Reply:
> ... but I don't have a clue how to manipulate the Makefile in order to change the sample-manifest.to use RUBY_CPU.

You can do that using `GENERATE_EXTRA_CMD`, for example.

Reply:
Okay, now the sample-manifest and the production one are mostly in sync, it published very well again  - thanks very much for the advises !